src/core.js

```diff
@@ -118,17 +118,18 @@ var alignWithDOM = function(nodeName, key, statics) {
     return;
   }
 
-  var existingNode = getChild(currentParent, key);
   var matchingNode;
 
+  if (key) {
+    matchingNode = getChild(currentParent, key);
+  }
+
   // Check to see if the node has moved within the parent or if a new one
   // should be created
-  if (existingNode) {
+  if (matchingNode) {
```

Comment:
We shouldn't have the logic flow changed like this for asserts. Maybe move this if into the if `(key)` and make it `matchingNode = matchingNode || createNode(...`.